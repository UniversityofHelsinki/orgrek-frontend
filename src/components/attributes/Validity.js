import React from 'react';
import { useTranslation } from 'react-i18next';
import { showValidity } from '../../actions/utilAction';

const Validity = ({ startDate, endDate }) => {
  const { t, i18n } = useTranslation();

  return (
    <span data-testid="validity">
      {showValidity(startDate, endDate, i18n, t)}
    </span>
  );
};

export default Validity;
