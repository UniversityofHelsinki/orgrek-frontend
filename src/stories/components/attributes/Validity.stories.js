import Validity from '../../../components/attributes/Validity';

export default {
  component: Validity,
};

export const Default = {
  args: {
    startDate: '2022-01-01',
    endDate: '2022-12-31',
  },
};

export const From = {
  args: {
    startDate: '2022-01-01',
    endDate: null,
  },
};

export const Until = {
  args: {
    startDate: null,
    endDate: '2022-12-31',
  },
};

export const Undefined = {
  args: {
    startDate: null,
    endDate: null,
  },
};
