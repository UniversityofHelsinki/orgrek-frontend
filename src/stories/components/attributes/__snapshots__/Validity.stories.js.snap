// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stories/components/attributes/Validity Default smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "1.1.2022 - 31.12.2022",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
  ],
  "name": "stories-components-attributes-validity--default",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`stories/components/attributes/Validity From smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "1.1.2022 from_date_react",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
  ],
  "name": "stories-components-attributes-validity--from",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`stories/components/attributes/Validity Undefined smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "not_specified",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
  ],
  "name": "stories-components-attributes-validity--undefined",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`stories/components/attributes/Validity Until smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "31.12.2022 until_date_react",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
  ],
  "name": "stories-components-attributes-validity--until",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
